### PR TITLE
NTRN-323 Feat: Add dex pause dao logic

### PR DESCRIPTION
**Pull request by jcompagni10**
Allow dao to change permissions for the dex. 

Related PRs:
https://github.com/neutron-org/neutronjsplus/pull/41
https://github.com/neutron-org/neutron-integration-tests/pull/313/files

**Integration test run**
https://github.com/neutron-org/neutron-tests/actions/runs/10012028532

### CODE REVIEW

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9589874469

**Comment by pr0n00gler** [comment]
Please attach a successful testrun

**Comment by jcompagni10** [comment]
> Please attach a successful testrun

It's in the description. 
https://github.com/neutron-org/neutron-tests/actions/runs/9589874469

Is test run as a comment preferable in the future?

**Review by pr0n00gler — APPROVED** [review]

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/9863521823

**Review by swelf19 — APPROVED** [review]

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/10012028532

File: contracts/dao/neutron-chain-manager/src/contract.rs [hunk 1]
```diff
@@ -338,6 +346,57 @@ pub fn get_tokenfactory_params(
...
+) -> Result<(), ContractError> {
+    let msg_update_params: MsgUpdateParamsDex =
+        serde_json_wasm::from_str(proposal.message.as_str())?;
+
+    let dex_update_param_permission = strategy
+        .get_dex_update_param_permission()
+        .ok_or(ContractError::Unauthorized {})?;
+
+    let dex_params = get_dex_params(deps, ParamsRequestDex {})?;
+    // TODO: do this through iteration instead of ifs for future proofing new params
```
[the remark this answers is not in the document]
Comment (jcompagni10) [reply contracts/dao/neutron-chain-manager/src/contract.rs:365]: I agree it's a bit annoying. But as far as I know, there is no easy way to iterate through a rust struct with arbitrary keys. We could do this by marshalling with serde but it starts to get pretty ugly. 
Also, if new dex params are added we would still have to update the contract to change the allow strategies and struct for retrieving dex params. So I don't think it's terrible

**Comment by jcompagni10** [comment]
https://github.com/neutron-org/neutron-tests/actions/runs/10012028532

**Pull request merged**